databases/week3/week3.sql

```diff
@@ -0,0 +1,285 @@
+INSERT INTO `reservation` (
+    `number_of_guests`,
+    `meal_id`,
+    `created_date`
+) VALUES (
+	3,
+    1,
+	'2019-12-27 10:15:00'
+);
+INSERT INTO `reservation` (
+    `number_of_guests`,
+    `meal_id`,
+    `created_date`
+) VALUES (
+	5,
+	2,
+	'2019-12-26 17:15:00'
+);
+INSERT INTO `reservation` (
+    `number_of_guests`,
+    `meal_id`,
+    `created_date`
+) VALUES (
+	6,
+	3,
+	'2019-11-26 08:15:00'
+);
+select * from reservation;
+-- Get a reservation with any id, fx 1
+select * from reservation
+where id = 2 ;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+update reservation set number_of_guests = 7
+where id = 2;
+
+-- Delete a reservation with any id, fx 1
+DELETE FROM reservation 
+WHERE id = 2;
+
+-- Review
+
+-- Get all reviews
+select * from review;
+
+-- Add reviews
+INSERT INTO `review` (
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+) VALUES (
+	'nice ambience',
+    'we were very happy with service and food',
+    2,
+    4,
+    DATE(NOW())
+);
+INSERT INTO `review` (
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+) VALUES (
+	'very good',
+    'we were very satisfied',
+    2,
+    5,
+    DATE(NOW())
+);
+INSERT INTO `review` (
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+) VALUES (
+	'average food quality',
+    'scope for improvement',
+    3,
+    2,
+    DATE(NOW())
+);
+INSERT INTO `review` (
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+) VALUES (
+	'slow service',
+    'rude waiters',
+    4,
+    1,
+    DATE(NOW())
+);
+
+-- Get a review with any id, fx 1
+SELECT * FROM review
+WHERE id = 1;
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE review
+SET title = 'awesome meal'
+WHERE id = 1;
+
+-- Delete a review with any id, fx 1
+DELETE FROM review 
+WHERE id = 1;
+
+-- Additional queries
+
+-- Get meals that has a price smaller than a specific price fx 90
+select * from meal;
+select * from meal where price < 90;
+
+-- Get meals that still has available reservations
+SELECT * FROM reservation 
+JOIN meal
+	ON meal.id = reservation.meal_id
+WHERE number_of_guests < max_reservation;
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+select * FROM meal
+WHERE title LIKE "%pizza%";
+
+-- Get meals that has been created between two dates
+select * from meal where (created_date BETWEEN '2019-12-20 ' AND '2019-12-30 ');
+
+-- Get only specific number of meals fx return only 5 meals
+select * from meal
+order by id limit 5;
+
+-- Get the meals that have good reviews
+
+SELECT meal.id, meal.title, meal.location, review.stars, review.title, review.description, review.created_date
+FROM meal
+INNER JOIN review ON review.meal_id = meal.id
+WHERE review.stars >= 4;
+        
```

Comment:
The above statement is meant to produce the average of the number of stars a meals has received
```

SELECT meal.id,
       meal.title,
       meal.location,
       Avg(review.stars) avg_stars,
       review.title,
       review.description,
       review.created_date
FROM   meal
       INNER JOIN review
               ON review.meal_id = meal.id
GROUP  BY review.meal_id
HAVING Avg(review.stars) > 3;  
```